source/unified-test-format/unified-test-format.rst

```diff
@@ -1336,6 +1363,76 @@ MUST assert that the field is not set, or, if the driver uses a nonpositive Int3
 value to indicate the field being unset, MUST assert that ``serverConnectionId``
 is a nonpositive Int32.
 
+expectedLogMessagesForClient
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A list of log messages that are expected to be observed (in that order) for a 
+client while executing `operations <test_operations_>`_.
+
+The structure of each object is as follows:
+
+- ``client``: Required string. Client entity for which the messages are expected
+  to be observed. See `commonOptions_client`_.
+
+- ``messages``: Required array of `expectedLogMessage`_ objects. List of 
+  messages, which are expected to be observed (in this order) on the corresponding
+  client while executing `operations`_. If the array is empty, the test runner
+  MUST assert that no messages were observed on the client. The driver MUST assert
+  that the messages produced are an exact match, i.e. that the expected and actual
+  message counts are the same and that there are no extra messages emitted by the
+  client during the test run.
+
+expectedLogMessage
+~~~~~~~~~~~~~~~~~~
+
+A log message which is expected to be observed while executing the test's
+operations.
+
+The structure of each object is as follows:
+
+- ``level``: Required string. This MUST be one of the level names listed in
+   `log severity levels <logging/logging.rst#log-severity-levels>`__. This
+   specifies the expected level for the log message and corresponds to the
+   level used for the message in the specification that defines it. Note that
+   since not all drivers will necessarily support all log levels, some driver
+   may need to map the specified level to the corresponding driver-supported
+   level. Test runners MUST assert that the actual level matches this value.
+
+- ``component``: Required string. This MUST be one of the component names listed
+   in `components <../logging/logging.rst#components>`__. This specifies the
+   expected component for the log message. Note that since naming variations
+   are permitted for components, some drivers may need to map this to a
+   corresponding language-specific component name. Test runners MUST assert
+   that the actual component matches this value.
+
+- ``failureIsRedacted``: Optional boolean. This field SHOULD only be specified
+  when ``data`` contains ``{ failure: { $$exists: true }}``; the test runner
+  MUST report an error if this is not the case.
+
+  When ``hasFailure`` is present and its value is ``true``,
+  the test runner MUST assert that the failure has been redacted according to
+  the rules defined for error redaction in the `command logging and monitoring
+  specification <../command-logging-and-monitoring/command-logging-and-monitoring.rst#security>`__.
+
+  When ``false``, the test runner MUST assert that the failure has NOT been
+  redacted.
```

Comment:
To my earlier comment, you could also just change the behavior for handling `true` and `false`. For instance, this could be:

> When `failureIsRedacted` is specified and ``false``, the test runner MUST assert that a failure is present and has NOT been redacted.

And likewise:

> When `failureIsRedacted` is specified and `true`, the test runner MUST assert that a failure is present and...

Reply:
done